File: examples/lambda/nodejs/handler.mjs
export async function handler(event, context = {}) {
  const numbers = Array.isArray(event.numbers) ? event.numbers : [];
  const sum = numbers.reduce((total, value) => total + Number(value || 0), 0);

  return {
    requestId: context.awsRequestId ?? event.requestId ?? "local",
    message: `Hello ${event.name ?? "microVM"}`,
    sum,
    runtime: "nodejs",
    timestamp: new Date().toISOString(),
  };
}

File: examples/lambda/python/handler.py
from __future__ import annotations

from datetime import datetime, timezone
from typing import Any


def handler(event: dict[str, Any], context: Any = None) -> dict[str, Any]:
    numbers = event.get("numbers") if isinstance(event.get("numbers"), list) else []
    total = sum(float(value or 0) for value in numbers)
    request_id = getattr(context, "aws_request_id", None) or event.get("requestId") or "local"

    return {
        "requestId": request_id,
        "message": f"Hello {event.get('name', 'microVM')}",
        "sum": total,
        "runtime": "python",
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }
